### PR TITLE
[Fix] : Handle non geospatial ogc services

### PR DESCRIPTION
**Pull request by cmoinier**
### Description

This PR fixes a bug found in MEL datahub, where the OGC API services that are non geospatial were wrongfully displayed on the map component.
In this PR, we test if one of the features of the service has or not a geometry. Depending on that, if it's the only service, the map will or will not be displayed. If there are other services viewable in the map, the OGC service will be removed from the dropdown list if not geospatial. 

### Quality Assurance Checklist

- [x] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [ ] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

---

**This work is sponsored by MEL**.

### CODE REVIEW

**Review by tkohr — COMMENTED** [review]
Thanks, @cmoinier, this works really well! I'm just noticing that the `map` tab is still clickable when there is a problem with the OGC API calls. This can easily be reproduced in dev by not using a CORS plugin in the browser. It's not directly related to this PR as it also causes an uncaught error on `main` I think. I propose to take a look if you can solve it here, and otherwise we'll solve it in a separate PR. Note there is [GSMEL-542](https://jira.camptocamp.com/browse/GSMEL-542) that seems related.

**Review by tkohr — COMMENTED** [review]
Thanks for the additional fix @cmoinier, which works well on my side!

Do you think you could reflect your changes for this in the unit tests of the `map-utils.service.ts`, `data.service.ts` and `mdview.facade.ts` by adapting fixtures for the fixed case and adding tests for new methods?

File: apps/datahub/src/app/record/record-metadata/record-metadata.component.ts [hunk 1]
```diff
@@ -22,12 +22,11 @@ export class RecordMetadataComponent {
 
   displayMap$ = combineLatest([
     this.metadataViewFacade.mapApiLinks$,
-    this.metadataViewFacade.geoDataLinks$,
+    this.metadataViewFacade.geoDataLinksWithGeometry$,
```
[the remark this answers is not in the document]
Comment (cmoinier) [reply apps/datahub/src/app/record/record-metadata/record-metadata.component.ts:25]: Done in next commit :+1:

**Review by tkohr — APPROVED** [review]
Nice, thanks for the added tests and the `falsy` start :star_struck:

**Pull request merged**